### ISSUE TITLE
almostequals fails on nested lists

### ISSUE THREAD

**Issue by elkins0** [CindyJS/CindyJS#279]
The `comp_notalmostequals`, `comp_almostequals`, and `List.almostequals` functions all fail on nested lists with an exception:
```
Object doesn't support property or method 'comp_almostequals'
```

1. The `comp_notalmostequals` function calls the `comp_almostequals`  function which calls the `List.almostequals` function which calls an undefined `List.comp_almostequals` function. The  `List.almostequals` function should call the `List.almostequals` function instead of `List.comp_almostequals`.
2. The parameter passing of modifs is inconsistent in both the `List.almostequals` and the `List.equals` functions. Both should receive and pass on the modifs parameter.
3. There are no test cases for nested lists for any of the failing functions.

**Comment by gagern** [comment]
1. That bug is fixed by #441.

2. Why should these internal functions receive the CindyScript modifiers? What would they use them for? In particular since the fuzzy comparisons are usually called using infix operators, and one cannot pas modifiers to those, I don't see a use case for this.

3. That is still true. I though I had done something along these lines for 4d7c929f779fe7fc16475a2e0f3cf8f5558de128 or #446, but apparently not. I guess I will add it to #446.

**Comment by elkins0** [comment]
@gagern "2.Why should these internal functions receive the CindyScript modifiers?"

I only see that the `modifs` parameter  is lost if `comp_almostequals` calls `List.almostequals` which could call `comp_almostequals` again. If that is not a possibility then there is no problem.

```javascript
function comp_almostequals(args, modifs) {
...
        var erg = List.almostequals(v0, v1); // modifs not passed
...
}

List.almostequals = function(a1, a2) {
...
            erg = erg && comp_almostequals([av1, av2], []).value; // modifs not available to be passed on
...
}
```

**Comment by gagern** [comment]
Thanks for the clarification, @elkins0! `comp_almostequals` doesn't make use of its `modifs` argument. That is listed only for the sake of consistency. So there is no harm in not passing it along. Thus 2 is not a bug, just slightly confusing code. With 1 addressed and a fix for 3 included in 1bd084bbcc139a810e983652cd550e9e2d796586 for #446, I think having #446 merged should close this one.

**Comment by elkins0** [comment]
@gagern OK by me to close. It's not an issue for me. But I might ask why the  `modifs`  parameter is passed into so many functions which fail to make any use of it and just clutter up the code. Unlike C++ and Java, JavaScript isn't choosing which function to call based on calling signatures. Are code syntax checkers forcing this calling pattern?

**Comment by gagern** [comment]
Personally I'm using this style because that's what the codebase has always been using. @richter-gebert introduced this style originally. I guess one main motivation might be copy and paste: if you want to create a new function or operator, you just copy the signature of an existing one and change the name. A second reason might be so that you don't have to change the signature if you add support for modifiers at some point. A third reason might be because it felt weird to pass a modifiers object on every invocation (as the interpreter does) without a positional parameter to accept it.

I guess that for the special case of infix operators (as opposed to named functions), it might make sense to drop them. Except that there are cases where there are both, a named function and an infix operator sharing the same implementation. So for the sake of consistency we use them everywhere.

**Comment by gagern** [comment]
Fixed via #446.
